redditBot.ts

```diff
@@ -0,0 +1,118 @@
+import Snoowrap from 'snoowrap';
+import { SubmissionStream } from 'snoostorm';
+import { SubmissionQueue } from './SubmissionQueue';
+import { getBase64Data, translate, uploadImage, tryParseUrl } from './translationUtils';
+
+const client = new Snoowrap({
+	userAgent: 'reddit-bot-ichigo-reader',
+	clientId: '[REDACTED]',
+	clientSecret: '[REDACTED]',
+	refreshToken: '[REDACTED]'
+});
+const pollTimeMs = 10 * 60 * 1000; // 10 minutes.
+const streamConfig = {
+	subreddit: 'testingground4bots',
+	results: 25,
+	frequency: pollTimeMs
+};
+
+const submissionAgeLimitMs = 24 * 60 * 60 * 1000; // 1 day.
+const queue = new SubmissionQueue(submissionAgeLimitMs);
+const submissionListener = new SubmissionStream(client, streamConfig);
+
+submissionListener.on('item', submission => {
+	queue.add(submission);
+});
+
+const timeoutMs = 20 * 60 * 1000; // 20 minutes.
+setInterval(processQueue, timeoutMs);
+
+async function processQueue() {
+	queue.clearExpired();
+	await queue.refresh();
+
+	for (const submission of queue) {
+		const result = await processQueueItem(submission);
```

Comment:
Originally this was a large for loop with a lot of stuff going on. You can greatly tame large for loops by moving their processing into a function like this.